File: frontend/src/types/environment.ts
export interface Environment {
  id: string
  name: string
  isActive: boolean
  version: number
  createdAt: string
  updatedAt: string
}

export interface Variable {
  id: string
  environmentId: string
  key: string
  value: string
  isSecret: boolean
  enabled: boolean
  sortOrder: number
  version: number
  createdAt: string
  updatedAt: string
}

File: internal/domain/entities/environment.go
package entities

import (
	"time"

	"github.com/google/uuid"
)

type Environment struct {
	ID          uuid.UUID
	WorkspaceID uuid.UUID
	Name        string
	IsActive    bool
	Version     int
	IsDelete    bool
	CreatedBy   string
	CreatedAt   time.Time
	UpdatedBy   string
	UpdatedAt   time.Time
}

type Variable struct {
	ID            uuid.UUID
	EnvironmentID uuid.UUID
	Key           string
	Value         string
	IsSecret      bool
	Enabled       bool
	SortOrder     int
	Version       int
	IsDelete      bool
	CreatedBy     string
	CreatedAt     time.Time
	UpdatedBy     string
	UpdatedAt     time.Time
}
